monkey/infection_monkey/monkey.py

```diff
@@ -103,6 +105,17 @@ def initialize(self):
                     "Default server: %s is already in command servers list" % self._default_server
                 )
 
+    @staticmethod
+    def change_working_dir():
+        # Change the current working directory to be the location of the monkey agent
+        # binary. This is particularly desirable when `data_dir` is set and the agent is
```

Comment:
I don't think I properly understand the purpose of this code. In production, the island always runs the agent binaries (not the source code), so the comment "Change the current working directory to be the location of the monkey agent binary." makes no sense, this code is already running from the location of the monkey agent binary, is it not? Also, we need to make sure all places, where binary references files by relative paths are tested but should be alright.

Reply:
If you look at the `run_local_monkey()` function, you'll see it runs the binary but does not change it's current working directory. The monkey binary is at `MONKEY_ISLAND_ABS_PATH/cc/binaries/`. I'm not sure what $PWD is, but it's not `MONKEY_ISLAND_ABS_PATH/cc/binaries/`. I think it's either `MONKEY_ISLAND_ABS_PATH` or `MONKEY_ISLAND_ABS_PATH/../~

https://github.com/guardicore/monkey/blob/d507e6f6172dd71de8d95c97544e72b304149984/monkey/monkey_island/cc/resources/local_run.py#L23-L53

Reply:
Investigate using `Popen`'s `cwd` parameter when the island executes the binary.

Reply:
@VakarisZ I used the `cwd` parameter and it seemed to work just fine. https://github.com/guardicore/monkey/pull/1101/commits/5a1a5fc6fe6aedce80650118890887d601d56729